pkg/operator/staticpod/controller/prune/prune_controller.go

```diff
@@ -0,0 +1,328 @@
+)
+
+// NewPruneController creates a new pruning controller
+func NewPruneController(
+	targetNamespace string,
+	podResourcePrefix string,
+	command []string,
+	kubeClient kubernetes.Interface,
+	operatorConfigClient common.OperatorClient,
+	eventRecorder events.Recorder,
+	dynamicClient dynamic.Interface,
+	operatorConfigGVR schema.GroupVersionResource,
+	operatorConfigName string,
+) *PruneController {
+	c := &PruneController{
+		targetNamespace:        targetNamespace,
+		podResourcePrefix:      podResourcePrefix,
+		command:                command,
+		failedRevisionLimit:    0,
+		succeededRevisionLimit: 0,
+
+		operatorConfigClient: operatorConfigClient,
+		kubeClient:           kubeClient,
+		eventRecorder:        eventRecorder,
+
+		dynamicClient:      dynamicClient,
+		operatorConfigGVR:  operatorConfigGVR,
+		operatorConfigName: operatorConfigName,
+
+		queue:            workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), "PruneController"),
+		prunerPodImageFn: getPrunerPodImageFromEnv,
+	}
+
+	operatorConfigClient.Informer().AddEventHandler(c.eventHandler())
+
+	return c
+}
+
+func (c *PruneController) syncRevisionLimits() error {
+	config, err := c.dynamicClient.Resource(c.operatorConfigGVR).Get(c.operatorConfigName, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+
+	revisionLimits, ok, err := unstructured.NestedMap(config.Object, "spec", "operatorRevisionLimit")
+	if err != nil {
+		return err
+	}
+	if !ok {
+		return nil
+	}
+
+	val, err := getRevisionLimit(revisionLimits, "failed")
+	if err != nil {
+		return err
+	}
+	c.failedRevisionLimit = val
+
+	val, err = getRevisionLimit(revisionLimits, "succeeded")
+	if err != nil {
+		return err
+	}
+	c.succeededRevisionLimit = val
+
+	return nil
+}
+
+func getRevisionLimit(revisionLimits map[string]interface{}, key string) (int, error) {
+	val, set := revisionLimits[key]
+	if !set {
+		return 0, nil
+	}
+
+	limit, ok := val.(int)
+	if !ok {
+		return 0, fmt.Errorf("error converting %s revision limit %v to int", key, val)
+	}
+	return limit, nil
+}
+
+func (c *PruneController) pruneRevisionHistory(operatorStatus *operatorv1.StaticPodOperatorStatus, resourceDir string) error {
+	var succeededRevisionIDs, failedRevisionIDs []int
+
+	configMaps, err := c.kubeClient.CoreV1().ConfigMaps(c.targetNamespace).List(metav1.ListOptions{})
+	if err != nil {
+		return err
+	}
+	for _, configMap := range configMaps.Items {
+		if !strings.HasPrefix(configMap.Name, statusConfigMapName) {
+			continue
+		}
+
+		if revision, ok := configMap.Data["revision"]; ok {
+			revisionID, err := strconv.Atoi(revision)
+			if err != nil {
+				return err
+			}
+			switch configMap.Data["phase"] {
+			case string(corev1.PodSucceeded):
+				succeededRevisionIDs = append(succeededRevisionIDs, revisionID)
+			case string(corev1.PodFailed):
+				failedRevisionIDs = append(failedRevisionIDs, revisionID)
+			}
+		}
+	}
+
+	sort.Ints(succeededRevisionIDs)
```

Comment:
check length.  If you have none, you don't need to prune anything and you should return early.